File: DataBinding/PageAndRDL/CSVDataSource/VB.NET/CSVDataSource/MainForm.vb
Imports System.IO
Imports System.Resources

Public Class MainForm

	Dim _resource As New ResourceManager(Me.GetType)
	ReadOnly _settingForNoHeaderFixed = _resource.GetString("NoHeaderFixed")
	ReadOnly _settingForHeaderExistsFixed = _resource.GetString("HeaderExistsFixed")

	' Loads and shows the report.
	Private Sub btnCSV_Click(sender As System.Object, e As System.EventArgs) Handles btnCSV.Click
		Const settingForNoHeaderDelimited As String = "ProductID,ProductName,SupplierID,CategoryID,QuantityPerUnit,UnitPrice,UnitsInStock,UnitsOnOrder,ReorderLevel,Discontinued"

		Dim connectionString = String.Empty
		If rbtnNoHeaderComma.Checked Then
			connectionString = String.Format("Path={0};Encoding={1};TextQualifier="";ColumnsSeparator=,;RowsSeparator=\r\n;Columns={2}",
											 My.Resources.PathToFileNoHeaderComma, My.Resources.CSVEncoding, settingForNoHeaderDelimited)
		ElseIf rbtnHeaderTab.Checked Then
			connectionString = String.Format("Path={0};Encoding={1};TextQualifier="";ColumnsSeparator=" & vbTab & ";RowsSeparator=\r\n;HasHeaders=True",
											 My.Resources.PathToFileHeaderTab, My.Resources.CSVEncoding)
		ElseIf rbtnHeader.Checked Then
			connectionString = String.Format("Path={0};Encoding={1};Columns={2};HasHeaders=True",
											 My.Resources.PathToFileHeader, My.Resources.CSVEncoding, _settingForHeaderExistsFixed)
		ElseIf rbtnNoHeader.Checked Then
			connectionString = String.Format("Path={0};Encoding={1};Columns={2}",
											 My.Resources.PathToFileNoHeader, My.Resources.CSVEncoding, _settingForNoHeaderFixed)
		End If

		Dim report = New PageReport(New FileInfo("..\..\Reports\StockList.rdlx"))
		Dim connectionProps = report.Report.DataSources(0).ConnectionProperties
		connectionProps.DataProvider = "CSV"
		connectionProps.ConnectString = connectionString
		arvMain.ReportViewer.LoadDocument(report.Document)
	End Sub

End Class

File: DataBinding/PageAndRDL/CSVDataSource/C#/CSVDataSource/MainForm.cs
using System;
using System.IO;
using System.Resources;
using System.Windows.Forms;

namespace GrapeCity.ActiveReports.Samples.CSVDataSource
{
	public partial class MainForm : Form
	{
		readonly string _settingForNoHeaderFixed;
		readonly string _settingForHeaderExistsFixed;
		public MainForm()
		{
			InitializeComponent();
			var _resource = new ResourceManager(GetType());
			_settingForNoHeaderFixed = _resource.GetString("NoHeaderFixed");
			_settingForHeaderExistsFixed = _resource.GetString("HeaderExistsFixed");
		}

		// Loads and shows the report.
		private void btnCSV_Click(object sender, EventArgs e)
		{
			const string settingForNoHeaderDelimited = "ProductID,ProductName,SupplierID,CategoryID,QuantityPerUnit,UnitPrice,UnitsInStock,UnitsOnOrder,ReorderLevel,Discontinued";

			var connectionString = string.Empty;
			if (rbtnNoHeaderComma.Checked)
				connectionString = string.Format(@"Path={0};Encoding={1};TextQualifier="";ColumnsSeparator=,;RowsSeparator=\r\n;Columns={2}",
										Properties.Resources.PathToFileNoHeaderComma, Properties.Resources.CSVEncoding, settingForNoHeaderDelimited);
			else if (rbtnHeaderTab.Checked)
				connectionString = string.Format(@"Path={0};Encoding={1};TextQualifier="";ColumnsSeparator=	;RowsSeparator=\r\n;HasHeaders=True",
					  Properties.Resources.PathToFileHeaderTab, Properties.Resources.CSVEncoding);
			else if (rbtnHeader.Checked)
				connectionString = string.Format(@"Path={0};Encoding={1};Columns={2};HasHeaders=True",
										Properties.Resources.PathToFileHeader, Properties.Resources.CSVEncoding, _settingForHeaderExistsFixed);
			else if (rbtnNoHeader.Checked)
				connectionString = string.Format(@"Path={0};Encoding={1};Columns={2}",
										Properties.Resources.PathToFileNoHeader, Properties.Resources.CSVEncoding, _settingForNoHeaderFixed);

			var report = new PageReport(new FileInfo(@"..\..\Reports\StockList.rdlx"));
			var connectionProps = report.Report.DataSources[0].ConnectionProperties;
			connectionProps.DataProvider = "CSV";
			connectionProps.ConnectString = connectionString;
			arvMain.ReportViewer.LoadDocument(report.Document);
		}
	}
}
